pkg/gencosmosdb/cosmosdb/template.go

```diff
@@ -196,11 +223,19 @@ func (i *templateQueryIterator) Next(ctx context.Context) (templates *pkg.Templa
 	} else {
 		headers.Set("X-Ms-Documentdb-Query-Enablecrosspartition", "True")
 	}
+	if i.options != nil && len(i.options.PartitionKeyRangeID) > 0 {
```

Comment:
is this not done by virtue of calling setOptions below?

Reply:
yes, it does